### PR TITLE
Make installation relocatable, and the installation testable

### PR DESCRIPTION
**Pull request by gezalore**
It makes the installation relocatable based on the discussion in #4893.

All real executables are installed in the pkgdatadir/bin, and a redirect script is installed in the bindir for the public executables.

I also fixed up the tests so they can be run against an installation with
`env VERILATOR_ROOT=path/to/share/verilator make -C path/to/test_regress`

The remaining 2 test failures are due to the .out files mismatching because of the change to test_regress/input.vc.

If you are happy with this otherwise I will fix those remaining 2 failures and add a CI job to test against a relocated installation.

### CODE REVIEW

**Comment by gezalore** [comment]
Sorry, forgot about the nodist/install_test earlier. Now fixed and passes.

File: bin/verilator [hunk 1]
```diff
@@ -34,6 +35,17 @@ my $opt_quiet_exit;
 my $opt_unlimited_stack = 1;
 my $opt_valgrind;
 
+my $verilator_root = realpath("$RealBin/..");
+if (defined $ENV{VERILATOR_ROOT}) {
+    if ((!-d $ENV{VERILATOR_ROOT}) || $verilator_root ne realpath($ENV{VERILATOR_ROOT})) {
+        warn "%Error: verilator: VERILATOR_ROOT is set to inconsistent path. Suggest leaving it unset.\n";
+        warn "%Error: VERILATOR_ROOT=$ENV{VERILATOR_ROOT}\n";
+        exit 1;
```
[the remark this answers is not in the document]
Comment (gezalore) [reply bin/verilator:43]: The intention here, is that if you invoke $FOO/bin/verilator, when you already have $VERILATOR_ROOT set, but $VERILATOR_ROOT is not the same as $FOO (that is, you are trying to run the wrapper script 'verilator' against a mismatching 'verailator_bin'), then Error.

The 'realpath' is there exactly to not fail if $VERILATOR_ROOT and $FOO are different but symlinked to the same place.

[the remark this answers is not in the document]
Comment (wsnyder) [reply bin/verilator]: Ok, I misread, you're comparing two realpaths.

**Comment by gezalore** [comment]
Added CI job to test the relocated installation. If it's all good then this can be merged now.

**Review by wsnyder — APPROVED** [review]
Suggest test reloc only on gcc, to reduce total CI time.

Feel free to merge once passing.

**Pull request merged**